### ISSUE TITLE
Logback Logs are not sending to Azure AppInsights

### ISSUE THREAD

**Issue by Rajkumar-dev** [microsoft/ApplicationInsights-Java#953]
### Expected behavior
Logback logs should send to application insights.

### Actual behavior
logs are not sending to AppInsights

### To Reproduce
1. Created Maven based Spring boot Project.
2. Added dependency of appinsights and logback to pom.xml
3. Added Application Insights properties along with instrumentation key under application.properties
4. Added aiappender in logback-spring.xml
5. create logger instance from logger factory.

Please find the above mentioned project under this link.
https://github.com/Rajkumar-dev/Java_AzureAppInsights_Logback_Springboot.git

**Comment by littleaj** [comment]
It looks like you need to add the `<instrumentationKey>` element into the logback config:
https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-trace-logs#add-the-appender-to-your-logging-framework

**Comment by Rajkumar-dev** [comment]
@littleaj 

I tried the same by adding instrumentationkey tag in logback-spring.xml. But it doesn't work out.

Could you please check that, am i missing something else.

Please find the project under this link.
https://github.com/Rajkumar-dev/Java_AzureAppInsights_Logback_Springboot.git

**Comment by Rajkumar-dev** [comment]
Hi

Anyone please check this issue

**Comment by littleaj** [comment]
I ran your app and had to do two things to make it work: 
1. Modify logback-spring.xml as such:
```xml
	<appender name="aiAppender" class="com.microsoft.applicationinsights.logback.ApplicationInsightsAppender">
		<instrumentationKey>c3cf9210-763f-4978-a782-bcfba91f2df6</instrumentationKey>
	</appender>
```
2. Add a sleep to let the application run longer than the buffer flush period.

I think what you were seeing is when the application shuts down, it was stopping the sender threads before they had a chance to flush. Short lived processes are not a supported scenario for the TelemetryClient at this time.

I verified this by turning on the sdk logger in application.properties:
```
azure.application-insights.logger.type=console
azure.application-insights.logger.level=trace
```

**Comment by Rajkumar-dev** [comment]
Hi @littleaj 

Thanks for your response. 

I have tried the same. When I am adding sleep at the end, it is able to flush the logging details to application insights.

So it means, At the end of the application if we have any logger details we can't send that to application insights. Am i right? is there any way to do immediateflush. I checked the same, immediateflush will work only for RollingFileappender, not for applicationinsights appender it seems, .

Please let me know your thoughts. If we are able to flush the same with out sleep it will be awesome.

**Comment by littleaj** [comment]
@Rajkumar-dev, yes, if you set the `max-telemetry-buffer-capacity` to 1 ([starter config properties](https://github.com/microsoft/ApplicationInsights-Java/tree/master/azure-application-insights-spring-boot-starter#configure-more-parameters-using-applicationproperties)), it will pass each item to the http client immediately. It's still possible for the sender's http client to terminate before the items are actually sent, but this will give it the best chance in a short-lived application.

**Comment by trask** [comment]
This is resolved in [3.0 Preview](https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-in-process-agent)
